### PR TITLE
add podman pod inspect

### PR DESCRIPTION
**Pull request by baude**
first pass of podman pod inspect

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: libpod/pod.go [hunk 1]
```diff
@@ -59,6 +59,20 @@ type podState struct {
 	CgroupPath string
 }
 
+// PodInspect represents the data we want to display for
+// podman pod inspect
+type PodInspect struct {
```
Comment (mheon) [inline libpod/pod.go]: These should probably go in pkg/inspect/inspect.go with the other structs from inspect

File: libpod/pod.go [hunk 2]
```diff
@@ -696,3 +710,35 @@ func (p *Pod) Status() (map[string]ContainerStatus, error) {
 // TODO add pod batching
 // Lock pod to avoid lock contention
 // Store and lock all containers (no RemoveContainer in batch guarantees cache will not become stale)
+
+// Inspect returns a PodInspect struct to describe the pod
+func (p *Pod) Inspect() (PodInspect, error) {
```
Comment (mheon) [inline libpod/pod.go]: Return a *PodInspect so we don't copy the struct on return

File: libpod/pod.go [hunk 3]
```diff
@@ -696,3 +710,35 @@ func (p *Pod) Status() (map[string]ContainerStatus, error) {
...
+			containerStatus = containerState.String()
+		}
+		pc := PodContainerInfo{
+			ID:    c.ID(),
+			State: containerStatus,
+		}
+		podContainers = append(podContainers, pc)
+	}
+	inspectData := PodInspect{
+		Config:     p.config,
```
Comment (mheon) [inline libpod/pod.go]: This is really not safe - we're passing pointers to the actual libpod structs outside of libpod, so people can modify them. Better to deepcopier them and return copies

File: libpod/runtime_pod.go [hunk 4]
```diff
@@ -129,3 +129,7 @@ func (r *Runtime) GetLatestPod() (*Pod, error) {
 	}
 	return pods[lastCreatedIndex], nil
 }
+
+func (p *Pod) GetConfig() *PodConfig {
+	return p.config
```
Comment (mheon) [inline libpod/runtime_pod.go]: As above, deepcopier before returning

File: cmd/podman/pod_inspect.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+		pod *libpod.Pod
+	)
+	args := c.Args()
+	runtime, err := libpodruntime.GetRuntime(c)
+	if err != nil {
+		return errors.Wrapf(err, "could not get runtime")
+	}
+	defer runtime.Shutdown(false)
+
+	if c.Bool("latest") {
```
Comment (haircommander) [inline cmd/podman/pod_inspect.go]: getPodsFromContext(cli.context, libpod.runtime) does this (in cmd/podman/utils.go)

File: cmd/podman/pod_inspect.go [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+	var (
+		pod *libpod.Pod
+	)
+	args := c.Args()
+	runtime, err := libpodruntime.GetRuntime(c)
+	if err != nil {
+		return errors.Wrapf(err, "could not get runtime")
+	}
+	defer runtime.Shutdown(false)
+
```
Comment (haircommander) [inline cmd/podman/pod_inspect.go]: We should check if both args and latest were both passed. checkMutuallyExclusiveFlags(cli.context) does this in cmd/podman/utils.go

File: libpod/pod.go (hunk 1 above)
Comment (mheon) [reply libpod/pod.go]: I can live without moving it, just wanted to do so if it was easy

File: libpod/pod.go [hunk 7]
```diff
@@ -696,3 +710,35 @@ func (p *Pod) Status() (map[string]ContainerStatus, error) {
...
+	var (
+		podContainers []PodContainerInfo
+	)
+
+	containers, err := p.AllContainers()
+	if err != nil {
+		return PodInspect{}, err
+	}
+	for _, c := range containers {
+		containerStatus := "uknown"
```
Comment (haircommander) [inline libpod/pod.go]: s/uknown/unknown ?

File: libpod/pod.go (hunk 3 above)
Comment (baude) [reply libpod/pod.go]: deepcopier it is!

File: libpod/pod.go (hunk 7 above)
Comment (haircommander) [reply libpod/pod.go]: this is the only blocking one, I am new to this github reviewing business.

File: cmd/podman/pod_inspect.go (hunk 5 above)
Comment (baude) [reply cmd/podman/pod_inspect.go]: done

File: libpod/pod.go (hunk 7 above)
Comment (baude) [reply libpod/pod.go]: fixed

File: cmd/podman/pod_inspect.go (hunk 6 above)
Comment (baude) [reply cmd/podman/pod_inspect.go]: done

File: docs/podman-pod-inspect.1.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
+% podman-pod-inspect "1"
+
+## NAME
+podman\-pod\-inspect- Displays information describing a pod
```
Comment (haircommander) [inline docs/podman-pod-inspect.1.md]: nit: space in between inspect and -
s/inspect-/inspect -

Comment (baude) [reply docs/podman-pod-inspect.1.md]: fixed

File: cmd/podman/pod_inspect.go [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/projectatomic/libpod/cmd/podman/libpodruntime"
+	"github.com/projectatomic/libpod/libpod"
+	"github.com/urfave/cli"
+)
+
+var (
+	podInspectFlags = []cli.Flag{
+		cli.BoolFlag{
```
[the remark this answers is not in the document]
Comment (baude) [reply cmd/podman/pod_inspect.go]: done

File: cmd/podman/pod_inspect.go (hunk 5 above)
Comment (baude) [reply cmd/podman/pod_inspect.go]: i mean i prefer my way :)

**Comment by haircommander** [comment]
LGTM assuming happy tests

**Comment by baude** [comment]
bot, retest this please

**Comment by baude** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
LGTM
@rh-atomic-bot r+

**Pull request closed without merging**